quinn-proto/src/connection/mod.rs

```diff
@@ -757,23 +769,35 @@ where
         })
     }
 
-    /// Returns `true` if a space has outgoing data to send
-    fn space_can_send(&self, space_id: SpaceId) -> bool {
-        if self.spaces[space_id].crypto.is_some() && self.spaces[space_id].can_send() {
-            return true;
+    /// Returns `if a space has outgoing data to send and what the expected type of data is
+    fn space_can_send(&self, space_id: SpaceId) -> SendableFrames {
+        if self.spaces[space_id].crypto.is_some() {
+            let can_send = self.spaces[space_id].can_send();
+            if !can_send.is_empty() {
+                return can_send;
+            }
         }
 
         if space_id != SpaceId::Data {
-            return false;
+            return SendableFrames::default();
```

Comment:
It wasn't immediately obvious to me what "default" meant here. Let's replace uses of it with a more explicit `SendableFrames::empty()` helper, for symmetry with `is_empty`.